tests/integration/devfile/cmd_dev_test.go

```diff
@@ -164,6 +166,51 @@ var _ = Describe("odo dev command tests", func() {
 			}, 180, 10).Should(Equal(true))
 
 		})
+	})
 
+	Context("port-forwarding for the component", func() {
+		When("devfile has single endpoint", func() {
+			BeforeEach(func() {
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+				helper.Cmd("odo", "project", "set", commonVar.Project).ShouldPass()
+				helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
+			})
+
+			It("should expose the endpoint on localhost", func() {
+				session := helper.CmdRunner("odo", "dev")
+				defer session.Kill()
+				helper.WaitForOutputToContain("Watching for changes in the current directory", 180, 10, session)
+
+				resp, _ := http.Get("http://localhost:40001")
+				defer resp.Body.Close()
+
+				body, _ := io.ReadAll(resp.Body)
+				helper.MatchAllInOutput(string(body), []string{"Hello from Node.js Starter Application!"})
```

Comment:
@feloy can you please help find a better way to assert things here? Problems from my perspective:
* Hard cording port 40001. Depending on individual perspective, this could either be an issue or be just fine
* Is `http.Get()` best way to assert here?

If you think this approach is just fine, let me know as well.

Reply:
- I would assert that the port 40001 is closed before to execute `odo dev`
- Yes, I think `http.Get` is what we need to test the connection (given the container speaks HTTP and not another protocol)

Reply:
> I would assert that the port 40001 is closed before to execute `odo dev`

Do you mean to assert this in the test or in the code? In the code, under the `IsPortFree` function, we have `err = listener.Close()` which closes the port.